### PR TITLE
W-11998991: Fix JSON validation when there are trailing tokens after …

### PR DESCRIPTION
**Pull request by giss-ignacio**
…first object

### CODE REVIEW

File: src/main/java/org/mule/module/json/internal/JsonSchemaValidator.java [hunk 1]
```diff
@@ -7,6 +7,7 @@
 package org.mule.module.json.internal;
 
 import static com.fasterxml.jackson.core.JsonParser.Feature.STRICT_DUPLICATE_DETECTION;
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_TRAILING_TOKENS;
```
Comment (acurci-at-mulesoft) [inline src/main/java/org/mule/module/json/internal/JsonSchemaValidator.java:10]: Please, correct the import order.

**Review by GuidoGianfrate — APPROVED** [review]

File: src/main/java/org/mule/module/json/internal/JsonSchemaValidator.java [hunk 2]
```diff
@@ -285,6 +288,11 @@ private JsonSchemaValidator(JsonSchema schema, ObjectMapper objectMapper) {
    * @param input the json to be validated
    */
   public void validate(InputStream input) {
+
+    if (Boolean.parseBoolean(System.getProperty(VALIDATOR_FAIL_ON_TRAILING_TOKENS))) {
```
Comment (acurci-at-mulesoft) [inline src/main/java/org/mule/module/json/internal/JsonSchemaValidator.java:292]: Shouldn’t you use System.getProperty(VALIDATOR_FAIL_ON_TRAILING_TOKENS, false) to set a default value?

File: src/test/java/org/mule/extension/BadObjectValidationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  @Override
+  protected void doSetUp() throws Exception {
+    json = doGetResource("inputs/bad-object.json");
+    System.setProperty(VALIDATOR_FAIL_ON_TRAILING_TOKENS, "true");
+  }
+
+  @Test
+  public void validate() throws Exception {
```
Comment (acurci-at-mulesoft) [inline src/test/java/org/mule/extension/BadObjectValidationTestCase.java:40]: Please, could you include the test with the system property on false to try the error case?

**Review by acurci-at-mulesoft — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/module/json/internal/JsonSchemaValidator.java (hunk 2 above)
Comment (giss-ignacio) [reply src/main/java/org/mule/module/json/internal/JsonSchemaValidator.java:292]: the getProperty only take String as parameters. And puts a default value when the property value is null. But the Boolean.parseBoolean(null) is false. So I think it's redundant. It should be:
`(Boolean.parseBoolean(System.getProperty(VALIDATOR_FAIL_ON_TRAILING_TOKENS, "false")))`

File: src/test/java/org/mule/extension/BadObjectValidationTestCase.java (hunk 3 above)
Comment (giss-ignacio) [reply src/test/java/org/mule/extension/BadObjectValidationTestCase.java:39]: yes

**Review by acurci-at-mulesoft — APPROVED** [review]

**Pull request merged**